R/ark.R

```diff
@@ -200,7 +233,13 @@ keep_open <- function(db_con, streamable_table, lines, p, tablename, con,
   while (TRUE) {
     p$tick()
     data <- DBI::dbFetch(res, n = lines)
-    if (nrow(data) == 0) break
+    
+    if (!is.null(callback))
+      data <- callback(data)
```

Comment:
Implementation is extremely straightforward df passthrough. This could have memory implications, but easily managed by controlling nlines.